### ISSUE TITLE
Feature: Create a rule to specify a version number for a specific artifact

### ISSUE THREAD

**Issue by rupakg** [hadolint/hadolint#692]
- [ ] This is a bug report
- [x] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

* Similar to these rules: [DL3007](https://github.com/hadolint/hadolint/wiki/DL3007) and [DL3008](https://github.com/hadolint/hadolint/wiki/DL3008), if we can create a new rule that can allow to check for a specific version for an artifact. The specific version can or may be specified by the custom `.hadolint.yaml` file.

### Steps to reproduce the behavior

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version` or
  `docker run --rm ghcr.io/hadolint/hadolint hadolint --version`:**

```bash
(paste your output here)
```

**Dockerfile** (if relevant)

```Dockerfile
(paste content of your Dockerfile)
```

**Additional environment details (OS, stack version, etc.)**

**Comment by lorenzo** [comment]
Can you show an example of how this would look like?

**Comment by bbugh** [comment]
I wasn't sure if it was worth creating a new issue, we want something like this. 

We have some internal base images (which themselves used pinned version) that we always want to use as latest for our other images. We can use `# hadolint ignore=DL3007` but that doesn't capture the intent, and it would be preferable to add something like:

```yaml
# .hadolint.yml
pinnedVersions:
  namespace/internal-image: latest
  namespace/other-base-image: latest
```

It could be done in other ways like adding rule-specific options, but there's no precedence for that that I could see, so this seemed more plausible.
